inc/vcf/parsing_state.hpp

```diff
@@ -40,6 +41,8 @@ namespace ebi
         
         std::shared_ptr<Source> source;
         std::shared_ptr<std::vector<Record>> records;
+        std::shared_ptr<std::vector<std::shared_ptr<Error>>> errors;
+        std::shared_ptr<std::vector<std::shared_ptr<Error>>> warnings;
```

Comment:
Do these need to be shared_ptr? Couldn't they be unique_ptr, since they only belong to the container?